src/Core/NotificationCenter/Entities/NotificationStatusDetails.cs

```diff
@@ -0,0 +1,26 @@
+﻿#nullable enable
+using System.ComponentModel.DataAnnotations;
+using Bit.Core.Enums;
+using Bit.Core.NotificationCenter.Enums;
+
+namespace Bit.Core.NotificationCenter.Entities;
+
+public class NotificationStatusDetails
+{
+    // Notification fields
+    public Guid Id { get; set; }
+    public Priority Priority { get; set; }
+    public bool Global { get; set; }
+    public ClientType ClientType { get; set; }
+    public Guid? UserId { get; set; }
+    public Guid? OrganizationId { get; set; }
+    [MaxLength(256)]
+    public string? Title { get; set; }
+    public string? Body { get; set; }
+    public DateTime CreationDate { get; set; }
+    public DateTime RevisionDate { get; set; }
+    // Notification Status fields
+    public Guid? NotificationStatusUserId { get; set; }
```

Comment:
❓ This is here just to aid in querying right? Can we restructure the join usage to avoid this being defined in the view and response objects? I find it confusing to be there when it would be set as the above `UserId`.

Reply:
What about the case of a notification that is targeted at an organization? 

```
Id : 61c208d3-74ae-4d5f-9c92-150ab941d262
ClientType: All
UserId: NULL
OrganizationId: 44674c0f-a6af-4c09-9c65-03ab89094764
NotificationStatusUserId: <target-userId>
ReadDate: 2025-09-18 19:00:52.0000000
DeletedDate: 2025-09-18 19:00:52.0000000
```

Reply:
I thought about that, but ... does the caller care? A user is calling this. We could arguably drop both user ID properties.

Reply:
True, in these cases it's implied that the notification is for the given user.

Reply:
I can drop it from the `NotificationStatusDetails.cs`. Like you Both said, it does not matter.
But it does from `NotificationStatusDetailsView` and `Notification_ReadByUserIdAndStatus`.
We need to filter on `NotificationStatus.UserId`, so we only return current user statuses. This needs to be done from stored procedure, so the view have to expose it.
We also need `Notification.UserId`, so we only return notifications that match current user (assuming it's not null).
Both are nullable. I just can't see how we can do this if we only have one or another (or combined with ISNULL, also does not work).
Alternatively I can get rid of the `NotificationStatusDetailsView`, that solves this problem.

Reply:
The easiest path forward might be to keep these in the model and view then, but I don't want them to be seen in the API response model at least.

Reply:
I think it's fine to delete the `NotificationStatusUserId` field from `NotificationStatusDetails.cs` - so the only occurrence is in SQL files.

Reply:
Fixed in 5c936d09